shell/platform/android/io/flutter/embedding/engine/FlutterJNI.java

```diff
@@ -873,8 +873,9 @@ private native void nativeDispatchSemanticsAction(
   @UiThread
   public void setSemanticsEnabled(boolean enabled) {
     ensureRunningOnMainThread();
-    ensureAttachedToNative();
-    nativeSetSemanticsEnabled(nativeShellHolderId, enabled);
+    if (isAttached()) {
```

Comment:
Hi @reidbaker  is there a good way to test this? 

Things i have tried:
1. make nativeSetSemanticsEnabled public and mock it using mockito.
2. I try creating a public method that just call `nativeSetSemanticsEnabled` and use spy to mock the instance. 

both of them throws UnSatisfiedLink error on `nativeSetSemanticsEnabled`. The only way I can think of is to extends the entire FlutterJNI and override method to do nothing, but I am wondering if that is preferred or is there any other way.

Reply:
Since this pr does not have a test I cant see what testing environment you are using. My guess is that if your test contains whatever dependency has nativeSetSemanticsEnabled then the issue is that your c code needs to run on an android device (or maybe with robolectric).

I also found this chromium documentation https://www.chromium.org/developers/testing/android-tests/testing-android-code-that-crosses-the-c-java-boundary/ that suggest that a spy is the correct solution.

Reply:
I still can't find a good way to use `spy` without trigger the error. I end up override entire class a mock out the part i don't need